python/cudf/cudf/core/multiindex.py

```diff
@@ -1020,26 +960,17 @@ def serialize(self):
         header["type-serialized"] = pickle.dumps(type(self))
         header["names"] = pickle.dumps(self.names)
 
-        header["source_data"], frames = self._source_data.serialize()
+        header["columns"], frames = column.serialize_columns(self._columns)
 
         return header, frames
 
     @classmethod
     def deserialize(cls, header, frames):
         names = pickle.loads(header["names"])
-
-        source_data_typ = pickle.loads(
-            header["source_data"]["type-serialized"]
-        )
-        source_data = source_data_typ.deserialize(
-            header["source_data"], frames
-        )
-
-        names = pickle.loads(header["names"])
-        return MultiIndex(names=names, source_data=source_data)
+        columns = column.deserialize_columns(header["columns"], frames)
+        return cls._from_data(dict(zip(names, columns)))
```

Comment:
Should we provide an easy on-ramp for a pickled dataframe in older version's of `cudf` for upto 1 version/release to be able to deserialize and remove the back-ward compatibility? Since pickle dump and load would be using these APIs.

Reply:
I've added back the old logic with a DeprecationWarning, let me know if that doesn't look right to you.